HISTORY.md

```diff
@@ -1,5 +1,56 @@
 ### mlpack ?.?.?
 ###### ????-??-??
+  * Migrate from boost tests to Catch2 framework (#2523), (#2584).
+
+  * Update NumFOCUS team voting responsibilities (#3071).
```

Comment:
I would only list code/documentation changes, I'm not even sure this file is part or should be part of a release.